vms/evm/subscriber.go

```diff
@@ -84,7 +117,87 @@ func (s *subscriber) forwardLogs() {
 			continue
 		}
 		s.log <- *messageInfo
+
+		// Update the database with the latest block height
+		// TODO: This should also be done in a separate goroutine, rather than waiting for warp messages to be processed
+		// TODO: Rather than updating the db when logs are received, we may want to consider updating the db when messages are successfully relayed
+		err = s.db.Put(s.chainID, []byte(database.LatestBlockHeightKey), []byte(strconv.FormatUint(msgLog.BlockNumber, 10)))
+		if err != nil {
+			s.logger.Error(fmt.Sprintf("failed to put %s into database", database.LatestBlockHeightKey), zap.Error(err))
+		}
+	}
+}
+
+func (s *subscriber) Initialize() error {
+	ethClient, err := ethclient.Dial(s.nodeRPCURL)
+	if err != nil {
+		return err
+	}
+
+	// Get the latest processed block height from the database.
+	heightData, err := s.db.Get(s.chainID, []byte(database.LatestBlockHeightKey))
```

Comment:
Should we handle the "file doesn't exist" error case here for the first initialization of the relayer?

We should clearly document the behavior in this case, but I think it likely makes the most sense to treat the case where the file doesn't exist as the first run of the relayer, and not relay any messages prior to it (since that would involve checking each message from the start of the chain).

Reply:
If `Initialize` fails for any reason, we simply continue with normal operation, opening subscriptions, relaying messages, etc. So if the file doesn't exist, we don't back-process any warp messages, and only process new incoming ones. OTOH, if the file _does_ exist and has `latestProcessedBlock: 1`, then yes, we'll deliver every warp message from genesis (though I'm sure `eth_getLogs` has some limits built in).

Reply:
right now when we do `db.Get` it looks to me if a file does not exist, we still return an error, and when the `relayer.NewRelayer` sees the errror it'll return and not continue processing. 

I think we should return a specific file does not exist error from `db.Get`, and then in `NewRelayer` if we get that specific error back we return without setting `latestSeenBlock` and going through `processFromHeight`

Reply:
something like 
```
	// Get the latest processed block height from the database.
	latestSeenBlockData, err := r.db.Get(r.sourceChainID, []byte(database.LatestSeenBlockKey))
	if err != nil {
                if err == FileDoesNotExist {
                      r.logger.Info("First run of relayer for chain ID %s, db file does not exist")
                      return &r, sub, nil 
                }
		r.logger.Warn("failed to get latest block from database", zap.Error(err))
		return nil, nil, err
	}
	latestSeenBlock, success := new(big.Int).SetString(string(latestSeenBlockData), 10)
	if !success {
		r.logger.Error("failed to convert latest block to big.Int", zap.Error(err))
		return nil, nil, err
	}

	// Back-process all warp messages from the latest seen block to the latest block
	// This will query the node for any logs that match the filter query from the stored block height,
	// and process the contained warp messages. If initialization fails, continue with normal relayer operation, but log the error.
	err = sub.ProcessFromHeight(latestSeenBlock)
	if err != nil {
		logger.Warn(
			"Encountered an error when processing historical blocks. Continuing to normal relaying operation.",
			zap.Error(err),
		)
	}

	return &r, sub, nil
```

Reply:
Nice catch. Fixed this now. In the case in which the file doesn't exist, we return the relayer instance, but skip the `sub.ProcessFromHeight` call.

Reply:
> though I'm sure eth_getLogs has some limits built in

That's a good call out. We should check this to make sure we handle any page limits properly.

Reply:
It seems like this limit is imposed by the deployment or client, and not by the method itself. Most services limit the number of blocks to 500-2000. I'd recommend limiting it to 200 as a conservative starting point. We can reevaluate this figure or add batched calls to accomodate larger ranges in the future, but IMO this is not an immediate concern.